features/FEATURE_LWIP/lwip-interface/lwip_stack.c

```diff
@@ -651,18 +726,29 @@ nsapi_error_t mbed_lwip_bringup_2(bool dhcp, bool ppp, const char *ip, const cha
     if (ppp) {
        err_t err = ppp_lwip_connect();
        if (err) {
+           lwip_connected = NSAPI_STATUS_DISCONNECTED;
+           if (lwip_client_callback) {
+               lwip_client_callback(lwip_status_cb_handle, NSAPI_EVENT_CONNECTION_STATUS_CHANGE, NSAPI_STATUS_DISCONNECTED);
+           }
            return mbed_lwip_err_remap(err);
        }
     }
 
+
     if (!netif_is_link_up(&lwip_netif)) {
```

Comment:
For non-blocking case link up has to be waited before continuing to netif_set_up and to dhcp negotiation.